index.d.ts

```diff
@@ -2117,6 +2117,8 @@ declare module 'mongoose' {
 
   type UnpackedIntersection<T, U> = T extends (infer V)[] ? (V & U)[] : T & U;
 
+  type ProjectionFields<DocType> = {[Key in keyof Omit<LeanDocument<DocType>, '__v'>]?: number} & Record<string, number>;
```

Comment:
I don't like restricting projections to numbers. There are plenty of codebases out there that use true/false instead of 0/1. And MongoDB does also allow using strings in `select()`, although that has [different behavior than using true/false or 0/1](https://thecodebarbarian.com/whats-new-in-mongoose-5-13-sanitizeprojection.html).

Ideally I'd say this should be `[Key in keyof Omit<LeanDocument<DocType>, '__v'>]?: any`, but if you want to be more strict than that, we need to support:

1. Boolean
2. String
3. [Projection operators](https://docs.mongodb.com/manual/reference/operator/projection/)

Reply:
Thank you for your review. I'll follow your version with `any`.